### PR TITLE
fix: remove prisma client error

### PR DESCRIPTION
**Pull request by PiotrekPKP**
This solution worked in one of my project so I think it should be alright.
I still recommend checking it

### CODE REVIEW

File: packages/db/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "dev": "prisma studio --port 5556",
     "clean": "rm -rf .turbo node_modules",
     "db-push": "prisma db push",
-    "db-generate": "prisma generate"
+    "db-generate": "prisma generate && ./fix-prisma.sh"
```
Comment (juliusmarminge) [inline packages/db/package.json:11]: Is this the 'proper way' to run this? I saw Rox's solution saying he added something to the `turbo.json`

https://github.com/prisma/prisma/issues/10433#issuecomment-1264687986

File: packages/db/fix-prisma.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+echo "Fixing Prisma client..."
+
+sed -i 's/this.datamodel = import_fs6.default.readFileSync(config2.datamodelPath, "utf-8");/const myPath = config2.generator.output.value + "\/schema.prisma";this.datamodel = import_fs6.default.readFileSync(myPath, "utf-8");/g' ./.generated/client/runtime/index.js
```
Comment (TheoBr) [inline packages/db/fix-prisma.sh:3]: This sed script won't work on mac, since mac expects an extra string in front 😅

**Comment by juliusmarminge** [comment]
Deployment fails on some imports

image

**Comment by TheoBr** [comment]
Heads up this branch is gonna be hella out of date now that I reverted those changes, might be worth basing off of this: https://github.com/t3-oss/create-t3-turbo/pull/45

**Comment by PiotrekPKP** [comment]
I will close this pr as we're doing lots of changes here. I'll do follow up if needed ;)

**Comment by TheoBr** [comment]
Got really close to solving this in a private fork using git patches. Worked locally but not on vercel deploy. Can't easily share code so I'll rip snippets here:

`packages/db/patches/runtime-prisma.patch`
```diff
diff --git a/packages/db/.generated/client/runtime/index.js b/packages/db/.generated/client/runtime/index.js
index ad98783..70dc910 100644
--- a/packages/db/.generated/client/runtime/index.js
+++ b/packages/db/.generated/client/runtime/index.js
@@ -26949,7 +26949,8 @@ var LibraryEngine = class extends Engine {
   constructor(config2, loader = new DefaultLibraryLoader(config2)) {
     var _a3, _b2;
     super();
-    this.datamodel = import_fs6.default.readFileSync(config2.datamodelPath, "utf-8");
+    const path = config2.generator.output.value + "/schema.prisma";
+    this.datamodel = import_fs6.default.readFileSync(path, "utf-8");
     this.config = config2;
     this.libraryStarted = false;
     this.logQueries = (_a3 = config2.logQueries) != null ? _a3 : false;
```

`packages/db/package.json` (scripts)
```json
    "db-generate": "prisma generate && pnpm rewire-prisma-bs",
    "rewire-prisma-bs": "git apply ./patches/runtime-prisma.patch"
 ```

**Pull request closed without merging**